util/MultiplayerCommon.h

```diff
@@ -20,6 +21,132 @@ FO_COMMON_API extern const std::string SP_SAVE_FILE_EXTENSION;
 
 FO_COMMON_API extern const int ALL_EMPIRES;
 
+class GameRules;
+
+/////////////////////////////////////////////
+// Free Functions
+/////////////////////////////////////////////
+typedef void (*GameRulesFn)(GameRules&); ///< the function signature for functions that add Rules to the GameRules (void (GameRules&))
+
+/** adds \a function to a vector of pointers to functions that add Rules to
+  * the GameRules.  This function returns a boolean so that it can be used to
+  * declare a dummy static variable that causes \a function to be registered as
+  * a side effect (e.g. at file scope:
+  * "bool unused_bool = RegisterGameRules(&foo)"). */
+FO_COMMON_API bool RegisterGameRules(GameRulesFn function);
+
+/** returns the single instance of the GameRules class */
+FO_COMMON_API GameRules& GetGameRules();
+
+/** Database of values that control how the game mechanics function. */
+class FO_COMMON_API GameRules {
+public:
+    enum RuleType : int {
+        INVALID_RULE_TYPE = -1,
+        TOGGLE,
+        INT,
+        DOUBLE,
+        STRING
+    };
+
+    RuleType RuleTypeForType(bool dummy)
+    { return TOGGLE; }
+    RuleType RuleTypeForType(int dummy)
+    { return INT; }
+    RuleType RuleTypeForType(double dummy)
+    { return DOUBLE; }
+    RuleType RuleTypeForType(std::string dummy)
+    { return STRING; }
+
+    struct FO_COMMON_API Rule : public OptionsDB::Option {
+        Rule();
+        Rule(RuleType rule_type_, const std::string& name_, const boost::any& value_,
+             const boost::any& default_value_, const std::string& description_,
+             const ValidatorBase *validator_, bool engine_internal_,
```

Comment:
It tracks whether a rule is defined internally, ie. in C++ code, or externally, from a parsed rules definition. The C++ code knows that combat rounds is defined in itself (C++ code) so doesn't need to check that this is the case. Original purpose was to put the internal vs. scripted rules in separate tabs, but that has been replaced with an explicit category field. Now it's just tracking where they are defined, which isn't used anywhere, but probably will be incorporated into pedia listing of rules.